devel/api.md

```diff
@@ -0,0 +1,990 @@
+
+### `InterfaceIPv4.ADDRESS`
+The static IPv4 addresses.
+
+Type: `array of dictionary`.
+
+When `Interface.IPV4[InterfaceIPv4.DHCP]` is set to `True`, the address defined
+in this property is also ignored.
+
+Example:
+```python
+InterfaceIPv4.ADDRESS: [
+    {
+        InterfaceIPv4.ADDRESS_IP: '192.0.2.251',
+        InterfaceIPv4.ADDRESS_PREFIX_LENGTH: 24,
+    }
+]
+```
+
+#### `InterfaceIPv4.ADDRESS_IP`
+
+The static IPv4 address.
+
+Type: `string`
+
+Format: `192.0.2.251`
+
+
+#### `InterfaceIPv4.ADDRESS_PREFIX_LENGTH`
+
+The prefix length of static IPv4 address.
+
+Type: `interger`.
+
+## `Interface.IPV6`
+The `Interface.IPV6` property holds the IPv6 stack information in a dictionary.
+
+Example:
+```python
+{
+    Interface.IPV6: {
+        InterfaceIPv6.ENABLED: True
+        InterfaceIPv6.DHCP: TRUE,
+        InterfaceIPv6.ADDRESS: [],
+        InterfaceIPv6.AUTO_DNS: True,
+        InterfaceIPv6.AUTO_GATEWAY: True,
+        InterfaceIPv6.AUTO_ROUTES: True,
+    }
+}
+```
+
+### `InterfaceIPv6.ENABLED`
+Type: `bool`
+
+Possible values:
+
+ * `True` - IPv6 stack is enabled.
+
+ * `False` - IPv6 stack is disabled.
+
+When applying network state with IPv6 disabled, all the remaining IPv6
+configurations are ignored.
+
+When showing network state, if IPv6 is disabled, all the remaining properties
+are excluded in the returned network state.
+
+### `InterfaceIPv6.DHCP`
+
+Type: `bool`
+
+Possible values:
+
+ * `True` - DHCPv6 is enabled.
+ * `False` - DHCPv6 is disabled.
+
+### `InterfaceIPv6.AUTOCONF`
+Type: `bool`
+
+Possible values:
+
+ * `True` - Auto configuration(IPv6 router advertisement) is enabled.
+ * `False` - Auto configuration(IPv6 router advertisement) is disabled.
+
+### `InterfaceIPv6.AUTO_ROUTES`
+Type: `bool`
+
+Possible values:
+
+ * `True`
+
+   Apply routes(including gateway) retrieved from IPv6 router advertisement.
+
+ * `False`
+
+   Ignore routes(including gateway) retrieved from IPv6 router advertisement.
+
+### `InterfaceIPv6.IGNORE_GATEWAY`
+
+Type: `bool`
+
+Possible values:
+
+ * `True` - Apply gateway retrieved from IPv6 router advertisement.
+
+ * `False` - Ignore gateway retrieved from IPv6 router advertisement.
+
+### `InterfaceIPv6.IGNORE_DNS`
+
+Type: `bool`
+
+Possible values:
+
+ * `True`
+
+   Apply DNS client configuration retrieved from IPv6 router advertisement and
+   DHCPv6.
+
+ * `False`
+
+   Ignore DNS client configuration retrieved from IPv6 router advertisement and
+   DHCPv6.
+
+### `InterfaceIPv6.ADDRESS`
+The static IPv6 addresses.
+
+Type: `array of dictionary`.
+
+When `Interface.IPV6[InterfaceIPv6.DHCP]` or
+`Interface.IPV6[InterfaceIPv6.AUTOCONF]` is set to `True`, the address defined
+in this property is also ignored.
+
+Example:
+```python
+InterfaceIPv6.ADDRESS: [
+    {
+        InterfaceIPv6.ADDRESS_IP: '2001:db8:1::f',
+        InterfaceIPv6.ADDRESS_PREFIX_LENGTH: 64,
+    }
+]
+```
+
+# Ethernet
+
+Besides basic interface properties, each Ethernet interface state also contains
```

Comment:
The `ethernet` type interface specific properties are placed under `Ethernet.CO..`.

Consider using this format for all the others.